src/app/client/src/app/modules/program/components/program/program.component.ts

```diff
@@ -207,7 +207,7 @@ export class ProgramComponent implements OnInit, OnDestroy, AfterViewInit {
       if (this.tabs && this.programDetails.userDetails) {
         this.defaultView = _.find(this.tabs, { 'index': this.getDefaultActiveTab() });
         this.programStageService.addStage(this.defaultView.onClick);
-        this.component = this.componentMapping[this.defaultView.onClick];
+        this.component = this.programComponentsService.getComponentInstance(this.defaultView.onClick);
```

Comment:
Remove the componentMapping object if we are not using, which is defined as public.